File: Intents/app/src/main/java/com/alexmercerind/myapplication/MainActivity.java
package com.alexmercerind.myapplication;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.google.android.material.card.MaterialCardView;


class MyHolder extends RecyclerView.ViewHolder {
    public TextView textView;
    public MaterialCardView materialCardView;
    public MyHolder(View itemView) {
        super(itemView);
        this.textView = (TextView)itemView.findViewById(R.id.ListTileTextView);
        this.materialCardView = (MaterialCardView)itemView.findViewById(R.id.ListTile);
        this.materialCardView.setOnClickListener(
                (View view) -> {
                    Intent intent = new Intent(
                            itemView.getContext(),
                            ChildActivity.class
                    );
                    intent.putExtra("Text", this.textView.getText().toString());
                    view.getContext().startActivity(intent);
                }
        );
    }

}


class MyAdapter extends RecyclerView.Adapter<MyHolder> {

    @Override
    public MyHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(
                R.layout.list_tile,
                parent,
                false
        );
        return new MyHolder(view);
    }

    @Override
    public void onBindViewHolder(MyHolder holder, int position) {
        holder.textView.setText(String.format("This is %dth item in the RecyclerView.", position + 1));
    }

    @Override
    public int getItemCount() {
        return 10;
    }
}


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        RecyclerView recyclerView = (RecyclerView)findViewById(R.id.ListView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(new MyAdapter());
    }
}

File: Affirmations/app/src/main/java/com/alexmercerind/affirmations/MainActivity.kt
package com.alexmercerind.affirmations

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.view.children
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.card.MaterialCardView

class ItemAdapter(var context: Context, var affirmations: Array<String>, var images: Array<Int>) : RecyclerView.Adapter<ItemAdapter.ItemHolder>() {
    class ItemHolder(var view: View): RecyclerView.ViewHolder(view) {
        val imageView: ImageView = view.findViewById(R.id.ListTileImageView)
        val textView: TextView = view.findViewById(R.id.ListTileTextView)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemHolder {
        val adapterLayout = LayoutInflater.from(parent.context)
            .inflate(R.layout.list_tile, parent, false)
        return ItemHolder(adapterLayout)
    }

    override fun onBindViewHolder(holder: ItemHolder, position: Int) {
        holder.imageView.setImageResource(images[position])
        holder.textView.text = affirmations[position]
    }

    override fun getItemCount(): Int {
        return affirmations.size
    }
}

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val recyclerView = findViewById<RecyclerView>(R.id.RecyclerView)
        recyclerView.adapter = ItemAdapter(
            this,
            resources.getStringArray(R.array.affirmations),
            arrayOf<Int>(
                R.drawable.image1,
                R.drawable.image2,
                R.drawable.image3,
                R.drawable.image4,
                R.drawable.image5,
                R.drawable.image6,
                R.drawable.image7,
                R.drawable.image8,
                R.drawable.image9,
                R.drawable.image10
            )
        )
    }
}
